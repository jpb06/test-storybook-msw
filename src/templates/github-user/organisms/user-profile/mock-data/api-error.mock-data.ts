export const apiErrorMockData = {
  code: 500,
  message: 'Oh no!',
};
