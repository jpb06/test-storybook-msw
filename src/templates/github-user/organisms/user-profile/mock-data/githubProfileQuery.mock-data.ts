export const githubProfileQueryMockData = {
  info: {
    nickname: 'coolbro',
    email: 'cool.bro@yolo.org',
    name: 'Yolo McBro',
  },
  extra: {
    raw_info: {
      followers: 243,
      following: 345,
      public_repos: 12,
      public_gists: 1,
      location: 'Paris, France',
    },
  },
};
