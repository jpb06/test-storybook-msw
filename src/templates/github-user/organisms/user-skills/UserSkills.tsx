import { List, ListSubheader } from '@mui/material';
import React from 'react';

import { useSkillsQuery } from '@api';

import { useSkillsCategorySelection } from './hooks/useSkillsCategorySelection';
import { ListSkeleton } from './molecules/skeletons/ListSkeleton';
import { SkillsList } from './molecules/skills-list/SkillsList';

export const UserSkills = () => {
  const { isLoading, data: skillsCategories } = useSkillsQuery();
  const { openedCategoryId, handleCategoryClick } =
    useSkillsCategorySelection();

  if (isLoading) {
    return <ListSkeleton />;
  }

  return (
    <List
      sx={{ bgcolor: 'background.paper' }}
      component="nav"
      aria-labelledby="nested-list-subheader"
      subheader={
        <ListSubheader component="div" id="nested-list-subheader">
          Skills
        </ListSubheader>
      }
    >
      {skillsCategories?.map((category) => (
        <SkillsList
          {...category}
          key={category.id}
          openedCategoryId={openedCategoryId}
          onCategoryClicked={handleCategoryClick}
        />
      ))}
    </List>
  );
};
