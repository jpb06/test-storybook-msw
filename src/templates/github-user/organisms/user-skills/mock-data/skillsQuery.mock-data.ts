export const skillsQueryMockData = [
  {
    id: 1,
    name: 'Soft skills',
    skills: [
      { id: 6, name: 'Communication', idRoles: [3] },
      { id: 8, name: 'Information sharing', idRoles: [3] },
    ],
  },
  {
    id: 2,
    name: 'Management',
    skills: [
      {
        id: 7,
        name: 'Project drive',
        idRoles: [4],
      },
      { id: 9, name: 'Reporting', idRoles: [4] },
      { id: 11, name: 'Roadmap definition', idRoles: [4] },
    ],
  },
  {
    id: 3,
    name: 'Tech',
    skills: [
      { id: 1, name: 'jest', idRoles: [1, 4] },
      { id: 4, name: 'react', idRoles: [1, 4] },
      { id: 5, name: 'Typescript', idRoles: [1, 4] },
      { id: 10, name: 'Github actions', idRoles: [2] },
    ],
  },
];
