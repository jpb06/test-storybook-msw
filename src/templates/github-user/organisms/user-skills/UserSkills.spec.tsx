import { screen } from '@testing-library/react';

import { skillsQueryHandler } from '@api/msw-handlers';
import { render } from '@tests/renders/render';

import { UserSkills } from './UserSkills';
import { skillsQueryMockData } from './mock-data/skillsQuery.mock-data';

jest.mock('@logic/delay');

describe('UserProfile component', () => {
  describe('nominal case', () => {
    it('should match snapshot', async () => {
      skillsQueryHandler(skillsQueryMockData, 200, true);

      const { baseElement } = render(<UserSkills />);

      await screen.findByRole('navigation');

      expect(baseElement).toMatchSnapshot();
    });

    it('should display skills', async () => {
      skillsQueryHandler(skillsQueryMockData, 200, true);

      render(<UserSkills />);

      await screen.findByText(skillsQueryMockData[0].name);
      await screen.findByText(skillsQueryMockData[1].name);
      await screen.findByText(skillsQueryMockData[2].name);
    });
  });

  describe('error case', () => {
    it('should match snapshot', async () => {
      skillsQueryHandler({ code: 500, message: 'Oh no!' }, 500, true);

      const { baseElement } = render(<UserSkills />);

      await screen.findByRole('alert');

      expect(baseElement).toMatchSnapshot();
    });

    it('should display an error when request failed', async () => {
      skillsQueryHandler({ code: 500, message: 'Oh no!' }, 500, true);

      render(<UserSkills />);

      await screen.findByText(/we've been unable to load your skills/i);
      expect(screen.getByTestId('ErrorOutlineIcon')).toBeInTheDocument();
    });
  });
});
