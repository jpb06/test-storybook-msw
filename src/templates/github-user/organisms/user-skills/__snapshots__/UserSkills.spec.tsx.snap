// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile component error case should match snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-outlinedError MuiAlert-outlined css-1yc4i3e-MuiPaper-root-MuiAlert-root"
        role="alert"
      >
        <div
          class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
            data-testid="ErrorOutlineIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
          </svg>
        </div>
        <div
          class="MuiAlert-message css-acap47-MuiAlert-message"
        >
          We've been unable to load your skills
        </div>
      </div>
      <div
        class="MuiBox-root css-1k9ek97"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-bmcivo-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-u0ssf3-MuiSkeleton-root"
              style="width: 50px; height: 20px;"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-container css-17rbshw-MuiGrid-root"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-circular MuiSkeleton-wave css-k47ddw-MuiSkeleton-root"
                style="width: 30px; height: 30px;"
              />
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1dk1h2u-MuiSkeleton-root"
                style="width: 120px; height: 25px;"
              />
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-container css-17rbshw-MuiGrid-root"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-circular MuiSkeleton-wave css-k47ddw-MuiSkeleton-root"
                style="width: 30px; height: 30px;"
              />
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1dk1h2u-MuiSkeleton-root"
                style="width: 120px; height: 25px;"
              />
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-container css-17rbshw-MuiGrid-root"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-circular MuiSkeleton-wave css-k47ddw-MuiSkeleton-root"
                style="width: 30px; height: 30px;"
              />
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1dk1h2u-MuiSkeleton-root"
                style="width: 120px; height: 25px;"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`UserProfile component nominal case should match snapshot 1`] = `
<body>
  <div>
    <nav
      aria-labelledby="nested-list-subheader"
      class="MuiList-root MuiList-padding MuiList-subheader css-y7uye9-MuiList-root"
    >
      <div
        class="MuiListSubheader-root MuiListSubheader-gutters MuiListSubheader-sticky css-uob957-MuiListSubheader-root"
        id="nested-list-subheader"
      >
        Skills
      </div>
      <div
        class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="EmojiEventsIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5h-2V3H7v2H5c-1.1 0-2 .9-2 2v1c0 2.55 1.92 4.63 4.39 4.94.63 1.5 1.98 2.63 3.61 2.96V19H7v2h10v-2h-4v-3.1c1.63-.33 2.98-1.46 3.61-2.96C19.08 12.63 21 10.55 21 8V7c0-1.1-.9-2-2-2zM5 8V7h2v3.82C5.84 10.4 5 9.3 5 8zm14 0c0 1.3-.84 2.4-2 2.82V7h2v1z"
            />
          </svg>
        </div>
        <div
          class="MuiListItemText-root css-tlelie-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            Soft skills
          </span>
        </div>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ExpandMoreIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
      <div
        class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="EmojiEventsIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5h-2V3H7v2H5c-1.1 0-2 .9-2 2v1c0 2.55 1.92 4.63 4.39 4.94.63 1.5 1.98 2.63 3.61 2.96V19H7v2h10v-2h-4v-3.1c1.63-.33 2.98-1.46 3.61-2.96C19.08 12.63 21 10.55 21 8V7c0-1.1-.9-2-2-2zM5 8V7h2v3.82C5.84 10.4 5 9.3 5 8zm14 0c0 1.3-.84 2.4-2 2.82V7h2v1z"
            />
          </svg>
        </div>
        <div
          class="MuiListItemText-root css-tlelie-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            Management
          </span>
        </div>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ExpandMoreIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
      <div
        class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="EmojiEventsIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5h-2V3H7v2H5c-1.1 0-2 .9-2 2v1c0 2.55 1.92 4.63 4.39 4.94.63 1.5 1.98 2.63 3.61 2.96V19H7v2h10v-2h-4v-3.1c1.63-.33 2.98-1.46 3.61-2.96C19.08 12.63 21 10.55 21 8V7c0-1.1-.9-2-2-2zM5 8V7h2v3.82C5.84 10.4 5 9.3 5 8zm14 0c0 1.3-.84 2.4-2 2.82V7h2v1z"
            />
          </svg>
        </div>
        <div
          class="MuiListItemText-root css-tlelie-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            Tech
          </span>
        </div>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ExpandMoreIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
    </nav>
  </div>
</body>
`;
